test/external_project/find-package-diagnostics.cmake

```diff
@@ -0,0 +1,39 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+# list all search places
+# NOTE: this can be enabled globally by -DSEQAN3_EXTERNAL_PROJECT_FIND_DEBUG_MODE=1 to help debug search paths
+# set (CMAKE_FIND_DEBUG_MODE ${SEQAN3_EXTERNAL_PROJECT_FIND_DEBUG_MODE})
+
+message (STATUS "CMAKE_COMMAND: ${CMAKE_COMMAND}")
+message (STATUS "CMAKE_VERSION: ${CMAKE_VERSION}")
+message (STATUS "CMAKE_SYSTEM_NAME: ${CMAKE_SYSTEM_NAME}")
+
+message (STATUS "search paths: (see https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure)")
+message (STATUS "1) CMAKE_FIND_USE_PACKAGE_ROOT_PATH: ${CMAKE_FIND_USE_CMAKE_PATH}")
+message (STATUS "   SeqAn3_ROOT: ${SeqAn3_ROOT}")
+
+message (STATUS "2) CMAKE_FIND_USE_CMAKE_PATH: ${CMAKE_FIND_USE_CMAKE_PATH}")
+message (STATUS "   CMAKE_PREFIX_PATH: ${CMAKE_PREFIX_PATH}")
+message (STATUS "   CMAKE_FRAMEWORK_PATH: ${CMAKE_FRAMEWORK_PATH}")
+message (STATUS "   CMAKE_APPBUNDLE_PATH: ${CMAKE_APPBUNDLE_PATH}")
+
+message (STATUS "3) CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH: ${CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH}")
+message (STATUS "   SeqAn3_DIR: ${SeqAn3_DIR}")
+
+# 4) with hints
```

Comment:
```suggestion
# 4) with hints. See point 4 in https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure.
# There is no output.
```

Reply:
I also had trouble understanding this (even with the comments). I would suggest to be more verbose.

Reply:
Thank you :)